[Lim Zi Qiang Marcus] iP

marc-97 opened the pull request.

Commit:
Add Gradle support

Commit:
build.gradle: Update version to 8.29

Commit:
[LEVEL-1] Added greet, echo and exit

Commit:
Level-1: Add greet, echo and exit

Commit:
[LEVEL-2] Added add, list

Commit:
Level-2: Implement add, list function

Commit:
[LEVEL-3] Added mark as done

Commit:
Level-3: Add mark as done

Commit:
[LEVEL-4] Change add to support todos, events, deadlines

Commit:
Level-4: Change add to support todos, events, deadlines

Commit:
[A-TextUiTesting] added automated testing

Commit:
A-TextUiTesting: add automated testing

Commit:
Level-5: Add minimal error handling

Commit:
[LEVEL-5] Added minimal error handling

Commit:
[LEVEL-6] added delete function

Commit:
Level-6: add delete function

marc-97 committed:
Merge branch 'master' of https://github.com/marc-97/ip

marc-97 committed:
Level-5: Add more exception handlings

marc-97 committed:
Level-7: Implement save & load

marc-97 committed:
Level-8: Add minimal date format for event and deadline

marc-97 committed:
Merge branch 'branch-Level-7'

marc-97 committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Duke.java
#	src/main/java/Task.java

marc-97 committed:
A-MoreOOP: Extract Storage, TaskList, Ui to seperate classes

marc-97 committed:
A-MoreOOP: Add parser class, fix parser & text-ui-test

Modify text-ui-test to current version of Duke.
Fix parser issue where task name is not added properly for event and deadline.

marc-97 committed:
A-Packages: Package all classes accordingly

marc-97 committed:
A-JUnit: Add a few simple test cases

marc-97 committed:
A-JavaDoc: Add java documentations for all classes

marc-97 committed:
Level-9: Add find function for Duke

marc-97 committed:
Merge branch 'branch-A-JavaDoc'

marc-97 committed:
Merge branch 'branch-Level-9'

marc-97 committed:
Level-9 Add documentation for the list function

w2vgd commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,69 @@
+    }
+
+    /**
+     * Process msg and return the request (command) of the input.
```
Should `msg` be a standard word instead of a "texting-style" spelling as it is part of the javadoc and is meant for users to read? 
I noticed several other places with this issue too.

w2vgd commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,127 @@
+                    ui.printFormatted(ex.getMessage());
+                } catch (NumberFormatException ex) {
+                    ui.printFormatted("Please enter integer values..");
+                }
+            } else if (request.equals("todo")) {
+                try {
+                    Task task = Task.createTask(args, request, "", "");
+                    ui.printAdded(tasks, tasks.addTask(task));
+                    storage.save(tasks);
+                } catch (DukeException ex) {
+                    ui.printFormatted(ex.getMessage());
+                }
+            } else if (request.equals("deadline")) {
+                try {
+                    String[] deadStr = parser.getFormattedCommand();
+                    Task task = Task.createTask(deadStr[0], request, deadStr[1], deadStr[2]);
+                    ui.printAdded(tasks, tasks.addTask(task));
+                    storage.save(tasks);
+                } catch (DukeException ex) {
+                    ui.printFormatted(ex.getMessage());
+                }
+            } else if (request.equals("event")) {
+                try {
+                    String[] eventStr = parser.getFormattedCommand();
+                    Task task = Task.createTask(eventStr[0], request, eventStr[1], eventStr[2]);
+                    ui.printAdded(tasks, tasks.addTask(task));
+                    storage.save(tasks);
+                } catch (DukeException ex) {
+                    ui.printFormatted(ex.getMessage());
+                }
+            } else if (request.equals("find")) {
+                ui.printFound(tasks.findTask(args));
+            } else {
+                try {
+                    throwDK();
+                } catch (DukeException ex) {
+                    ui.printFormatted(ex.getMessage());
+                }
+            }
+        }
```
Perhaps try to avoid deep nesting and code duplication here by instead using an outer `try catch` block that catches the same `DukeException` that each `if` clause throws?

w2vgd commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,69 @@
+     * @return a string of request
+     */
+    public String getRequest() {
+        String[] tk = msg.split(" ");
+        return tk[0];
```
Perhaps use a more descriptive name other than `tk`? 
I noticed several other places using this variable name as well.

w2vgd commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,101 @@
+     * Return a LinkedList of <code>Task</code> by reading from a file.
+     * This method will read through the saved file and return a list of <code>Task</code>.
+     *
+     * @return
```
Should there be a description for the return value here?

w2vgd commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,101 @@
+                    } else {
+                        throw new DukeException("Save file is corrupted ): Will be creating a new file");
+                    }
+                    if (token[1].equals("1")) {
```
Perhaps avoid using a magic number and use a named constant for comparison here?

w2vgd commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,156 @@
+        if (type.equals("E") || type.equals("D")) {
+            return String.format("[%s][%s] %s (%s: %s)",
+                    type,
+                    done ? "X" : " ",
```
Perhaps make the code more obvious here by adding parentheses around the same grouping?

w2vgd commented on src/main/java/duke/task/TaskList.java:
```diff
@@ -0,0 +1,101 @@
+            throw new DukeException(
+                    String.format("Tried to delete nothing ????. (Size: %d | Task No: %d)", list.size(), pos));
+        }
+        Task task = this.list.remove(pos - 1);
```
Perhaps remove the redundant `this`?
I have noticed several places with this issue too.

w2vgd reviewed (commented):
Code is generally easy to read 👍 I have noticed a few coding standard violations and there are a few javadoc comments with missing descriptions. I have also recommended some potential changes to code quality that you can consider.

marc-97 committed:
A-CodingStandard: Correct some codes to suit standard

marc-97 committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/Duke.java

marc-97 committed:
Add basic java GUI from tutorial

maxxng commented on src/main/java/duke/exception/DukeException.java:
```diff
@@ -1,5 +1,8 @@
 package duke.exception;
 
+/**
+ * Custom exception class for Duke
```
Should there be a punctuation at the end?

maxxng commented on src/main/java/duke/task/Task.java:
```diff
@@ -4,20 +4,38 @@
+    /**
+     * Task constructor for todo.
+     *
+     * @param type
+     * @param name
```
Should the parameters require some description?

maxxng commented on src/main/java/duke/task/Task.java:
```diff
@@ -71,6 +126,19 @@ public String toSaveFormat() {
+     * @param type the type of the task
+     * @param date the date of the task (if applicable)
+     * @param preposition the preposition for the task (if applicable)
+     * @return
```
Should there be some comments in this line for the return value?

maxxng reviewed (commented):
In general, the javadoc comments written are clear and concise and while there are only a few violations, majority are good to go.

marc-97 committed:
Merge remote-tracking branch 'origin/add-gradle-support'

marc-97 committed:
Add gradle support

marc-97 committed:
Merge branch 'branch-A-Gradle'

marc-97 committed:
Add checkstyle xml files from AB3

marc-97 committed:
Merge branch 'branch-A-CheckStyle'

marc-97 committed:
Move input reading from Duke class to Ui class

marc-97 committed:
Fix code to adhere to java standards for classes in main

marc-97 committed:
Merge commit '3b19ba1790d6035a9e246f5d4ae627bf68778606' into branch-JavaFX-Tutorial

nowknowing commented on src/test/java/duke/DukeTest.java:
```diff
@@ -0,0 +1,77 @@
+            assertEquals("[T][ ] Read a book", task.toString());
+            assertEquals("T | 0 | Read a book", task.toSaveFormat());
+        } catch (DukeException ex) {
+            assertEquals(0, 1); // force fail
```
I think this is one way. But there are also ways to assert for exceptions. Alternatively you can simply 
public void todoTest() throws DukeException() {...}
I think

nowknowing reviewed (commented):
Nice readable code! Maybe you can consider moving the logic out of duke.

marc-97 committed:
Complete javafx tutorial

marc-97 committed:
Add GUI to Duke

marc-97 committed:
Merge branch 'branch-Level-10'

marc-97 committed:
Improve OOP by adding command and task classes

marc-97 committed:
Add assertions for Parser & Storage class

marc-97 committed:
Improve code quality
Add guard clause for Storage class, improve readability for Deadline, Event & Task class

marc-97 committed:
Merge pull request #2 from marc-97/branch-A-Assertions

Add assertions for Parser & Storage class

marc-97 committed:
Merge branch 'master' of https://github.com/marc-97/ip into branch-A-CodeQuality

# Conflicts:
#	src/main/java/duke/storage/Storage.java

marc-97 committed:
Merge pull request #3 from marc-97/branch-A-CodeQuality

Branch a code quality

marc-97 committed:
Add sorting of tasks by date

marc-97 committed:
Merge pull request #4 from marc-97/branch-C-Sort

Add sorting of tasks by date

marc-97 committed:
Improve GUI
Add rounded corners for display picture, send icon and background image for scrolling pane

marc-97 committed:
Add title and icons for GUI

marc-97 committed:
Update user guide

marc-97 committed:
Update user guide anchors

marc-97 committed:
Add UI screenshot

marc-97 committed:
Fix typo for user guide
Add a new line after list
